task7/D.py

```diff
@@ -0,0 +1,39 @@
+# Dependency inversion principle
```

Comment:
Your interpretation looks similar to "**Interface segregation principle**".
Could you clarify implementation?